src/status_im2/contexts/chat/messages/content/view.cljs

```diff
@@ -0,0 +1,107 @@
+      [quo/author
+       {:profile-name   display-name
+        :chat-key       from
+        :time-str       (time/timestamp->time timestamp)
+        :contact?       added?
+        :verified?      ens-verified
+        ;; TODO not implemented yet
+        :nickname       nil
+        :ens-name       nil
+        :untrustworthy? nil}])))
+
+;;;; SYSTEM MESSAGES
+(defn system-message-content [{:keys [content-type quoted-message] :as message-data}]
+  (if quoted-message
+    [pin/system-message quoted-message] ;;TODO (flexsurfer) why we detect pin-message with quoted-message param?
+    (case content-type
+      ;; TEXT
+      constants/content-type-system-text [system.text/text-content message-data]
+      ;; COMMUNITY TODO (flexsurfer) refactor and move to status-im2
+      constants/content-type-community [old-message/community message-data]
+      ;; CONTACT REQUEST  TODO (flexsurfer) refactor and move to status-im2
+      constants/content-type-contact-request [old-message/system-contact-request message-data])))
+
+;;;; USER MESSAGES
+(defn user-message-content [{:keys [content-type] :as message-data}]
+  [rn/touchable-highlight {:underlay-color (colors/theme-colors colors/neutral-5 colors/neutral-90)
+                           :style          {:border-radius 16}
+                           :on-press       #()
+                           :on-long-press  #(rf/dispatch [:bottom-sheet/show-sheet
+                                                          {:content (drawers/reactions-and-actions message-data)}])}
+   [rn/view {:padding-horizontal 12 :padding-vertical 8 :flex-direction :row}
+    ;; AVATAR
+    [avatar message-data]
+    [rn/view {:margin-left 8}
+     ;; AUTHOR NAME
+     [author message-data]
+     (case content-type
+       ;; TEXT TODO (flexsurfer) refactor and move to status-im2
+       constants/content-type-text [content.text/text-content message-data]
+       ;; EMOJI TODO (flexsurfer) refactor and move to status-im2
+       constants/content-type-emoji [old-message/emoji message-data]
+       ;; STICKER TODO (flexsurfer) refactor and move to status-im2
+       constants/content-type-sticker [old-message/sticker message-data]
+       ;; IMAGE TODO (flexsurfer) refactor and move to status-im2
+       constants/content-type-image [old-message/message-content-image message-data]
+       ;; AUDIO TODO (flexsurfer) refactor and move to status-im2
+       constants/content-type-audio [old-message/audio message-data]
+       ;; UNKNOWN
+       [content.unknown/unknown-content message-data])
+     [status/status message-data]]]])
+
+(defn message-with-reactions [{:keys [pinned pinned-by mentioned in-pinned-view? content-type
+                                      last-in-group? deleted? deleted-for-me? message-id]
+                               :as   message-data}
+                              {:keys [chat-id]}]
+  (if (or deleted? deleted-for-me?)
+    ;; DELETED
```

Comment:
I think we should generally avoid writing comments saying WHAT is being done, and instead use comments only to explain the HOW/WHY. If the WHAT is not clear enough, then renaming the binding/function or extracting a piece of code to a binding/function seems better for maintainability.

I would say we should remove these comments because the code is clear enough.

Reply:
these comments are mostly for faster navigation through code, so I highlight most major components in the the view, it's hard without them

Reply:
but if its opposite to others i'll remove them im fine

Reply:
I understand the reasons, but other developers have other means of navigating and understanding the structure of the code.

I can only speak for myself @flexsurfer, but in case it helps:

- I can always jump back and forth between function implementations with a single keybinding.
- I can fold/unfold any Clojure data structure (or Hiccup component) with a single keybinding.
- I can narrow/widen the code in certain pieces of code (not sure if this is a Emacs only thing).

When combined, these approaches help me not rely on comments. I do love section comments (with 3 or 4 `;`), because Emacs allows me to collapse and easily navigate sections.